src/application-insights/azext_applicationinsights/_params.py

```diff
@@ -0,0 +1,39 @@
+        c.argument('application', options_list=['--app, -a'], help='GUID or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal')
+
+    with self.argument_context('monitor app-insights events list') as c:
+        c.argument('application', options_list=['--app, -a'], help='GUID or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal')
+        c.argument('event_type', options_list=['--type'], help='The type of events to retrieve.')
+        c.argument('timespan', options_list=['--timespan', '-t'], help='The timespan over which to retrieve metric values. This is an ISO8601 time period value. If timespan is omitted, a default time range of `PT12H` ("last 12 hours") is used. The actual timespan that is queried may be adjusted by the server based. In all cases, the actual time span used for the query is included in the response.')
+
+    with self.argument_context('monitor app-insights events show') as c:
+        c.argument('application', options_list=['--app, -a'], help='GUID or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal')
+        c.argument('event_type', options_list=['--type'], help='The type of events to retrieve.')
+        c.argument('event', options_list=['--event'], help='GUID of the event to retrieve. This could be obtained by first listing and filtering events, then selecting an event of interest.')
+        c.argument('timespan', options_list=['--timespan', '-t'], help='The timespan over which to retrieve metric values. This is an ISO8601 time period value. If timespan is omitted, a default time range of `PT12H` ("last 12 hours") is used. The actual timespan that is queried may be adjusted by the server based. In all cases, the actual time span used for the query is included in the response.')
+
+    with self.argument_context('monitor app-insights query') as c:
+        c.argument('application', options_list=['--app, -a'], help='GUID or fully-qualified Azure resource name of Application Insights component. The application GUID may be acquired from the API Access menu item on any Application Insights resource in the Azure portal')
+        c.argument('analytics_query', help='Query to execute over Application Insights data.')
+        c.argument('timespan', options_list=['--timespan', '-t'], help='Timespan over which to query. Defaults to querying all available data.')
+        c.argument('applications', options_list=['--extra-apps'], nargs='+', help='Additional applications over which to union data for querying. Specify additional workspace IDs separated by space.')
```

Comment:
I just thought of this, instead of telling users to specify both a single main application as well as a separate list of applications, would it be better to just have one `applications` parameter since the end result is simply an union of all the applications specified?

Reply:
How do you want to reconcile this with specifying app + resource group multiple times? I've actually run into a similar issue elsewhere. We support an operation that I would probably call `metrics list` in the CLI, basically a POST version of the GET request behind `metrics show`. The REST API looks like: 
`POST https://api.applicationinsights.io/v1/apps/APP_GUID/metrics`
```json
[
  {
    "id": "failed",
    "parameters": {
      "metricId": "requests/failed",
      "timespan": "P2D",
      "interval": "P1D"
    }
  },
  {
    "id": "count",
    "parameters": {
      "metricId": "requests/count",
      "timespan": "P2D",
      "interval": "P1D"
    }
  }
]
```
In the CLI, how should this look? My difficulty is unpacking the array of objects from a flat command:
```
az monitor app-insights metrics list (--metric requests/failed --label failed -t P2D --interval P1d) (--metric requests/count --label count -t P2D --interval P1d) 
```

We're pushing people away from this and potentially deprecating so I'm not including it, but it's the same kind of issue.